test/unit/dialects/mssql/query-generator.test.js

```diff
@@ -22,6 +24,53 @@ if (current.dialect.name === 'mssql') {
       });
     });
 
+    it('upsertQuery with falsey values', function () {
+      let testTable = this.sequelize.define(
+        'test_table',
+        {
+          Name: {
+            type: DataTypes.STRING,
+            primaryKey: true
+          },
+          Age: {
+            type: DataTypes.INTEGER
+          },
+          IsOnline: {
+            type: DataTypes.BOOLEAN,
+            primaryKey: true
+          }
+        },
+        {
+          freezeTableName: true,
+          timestamps: false
+        }
+      );
+
+      const insertValues = {
+        Name: 'Charlie',
+        Age: 24,
+        IsOnline: false
+      };
+
+      const updateValues = {
+        Age: 24
+      };
+
+      const whereValues = [
+        {
+          Name: 'Charlie',
+          IsOnline: false
+        }
+      ];
+
+      const where = {
+        [Op.or]: whereValues
+      };
+
+      // the main purpose of this test is to validate this does not throw
+      this.queryGenerator.upsertQuery('test_table', updateValues, insertValues, where, testTable);
```

Comment:
Just like the test below, assert against a SQL query

Reply:
@sushantdhiman this has been done.